streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsAssignmentScaleTest.java

```diff
@@ -0,0 +1,256 @@
+        configMap.put(InternalConfig.STREAMS_ADMIN_CLIENT, adminClient);
+        configMap.put(InternalConfig.ASSIGNMENT_ERROR_CODE, new AtomicInteger());
+        configMap.put(InternalConfig.NEXT_SCHEDULED_REBALANCE_MS, new AtomicLong(Long.MAX_VALUE));
+        configMap.put(InternalConfig.TIME, new MockTime());
+        configMap.put(InternalConfig.INTERNAL_TASK_ASSIGNOR_CLASS, taskAssignor.getName());
+        configMap.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, numStandbys);
+
+        final MockInternalTopicManager mockInternalTopicManager = new MockInternalTopicManager(
+            new MockTime(),
+            new StreamsConfig(configMap),
+            new MockClientSupplier().restoreConsumer,
+            false
+        );
+        partitionAssignor.configure(configMap);
+        EasyMock.replay(taskManager, adminClient, mainConsumer);
+
+        partitionAssignor.setInternalTopicManager(mockInternalTopicManager);
+
+        final Map<String, Subscription> subscriptions = new HashMap<>();
+        for (int client = 0; client < numClients; ++client) {
+            for (int i = 0; i < numThreadsPerClient; ++i) {
+                subscriptions.put(getConsumerName(i, client),
+                                  new Subscription(
+                                      topic,
+                                      getInfo(uuidForInt(client), EMPTY_TASKS, EMPTY_TASKS).encode())
+                );
+            }
+        }
+
+        final long firstAssignmentStartMs = System.currentTimeMillis();
+        final Map<String, Assignment> firstAssignments = partitionAssignor.assign(clusterMetadata, new GroupSubscription(subscriptions)).groupAssignment();
+        final long firstAssignmentEndMs = System.currentTimeMillis();
+
+        final long firstAssignmentDuration = firstAssignmentEndMs - firstAssignmentStartMs;
+        if (firstAssignmentDuration > MAX_ASSIGNMENT_DURATION) {
+            throw new AssertionError("The first assignment took too long to complete at " + firstAssignmentDuration + "ms.");
+        } else {
+            log.info("First assignment took {}ms.", firstAssignmentDuration);
+        }
+
+        // Use the assignment to generate the subscriptions' prev task data for the next rebalance
+        for (int client = 0; client < numClients; ++client) {
+            for (int i = 0; i < numThreadsPerClient; ++i) {
+                final String consumer = getConsumerName(i, client);
+                final Assignment assignment = firstAssignments.get(consumer);
+                final AssignmentInfo info = AssignmentInfo.decode(assignment.userData());
+
+                subscriptions.put(consumer,
+                                  new Subscription(
+                                      topic,
+                                      getInfo(uuidForInt(i), new HashSet<>(info.activeTasks()), info.standbyTasks().keySet()).encode(),
+                                      assignment.partitions())
+                );
```

Comment:
Code style suggestion:
```suggestion
                subscriptions.put(
                    consumer,
                    new Subscription(
                        topic, 
                        getInfo(uuidForInt(i), new HashSet<>(info.activeTasks()), info.standbyTasks().keySet()).encode(), 
                        assignment.partitions()
                    )
                );
```